src/Chronicle.Integrations.EFCore/Persistence/EFCoreSagaLog.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Chronicle.Integrations.EFCore.Repositories;
+
+namespace Chronicle.Integrations.EFCore.Persistence
+{
+    internal class EFCoreSagaLog : ISagaLog
+    {
+        private readonly ISagaLogRepository SagaLogRepository;
+
+        public EFCoreSagaLog(ISagaLogRepository _sagaLogRepository)
+        {
+            SagaLogRepository = _sagaLogRepository ?? throw new ArgumentNullException(nameof(_sagaLogRepository));
+        }
+
+        public async Task<IEnumerable<ISagaLogData>> ReadAsync(SagaId id, Type type)
```

Comment:
async/await is not needed here since you can directly return the `Task` to the caller